accepted/future-releases/0546-patterns/feature-specification.md

```diff
@@ -2460,10 +2460,39 @@ All other types are not always-exhaustive. Then:
     other option is to throw an error and most Dart users prefer to catch those
     kinds of mistakes at compile time.*
 
-*   The language doesn't specify or mandate this, but implementations are
-    encouraged to report a warning if a case in a switch statement, collection
-    element, or expression is unreachable because all values it can match are
-    also matched by preceding cases.
+**Breaking change:** Currently, a non-exhaustive switch on an enum type is only
+a warning. This promotes it to an error. Also, switches on `bool` do not
+currently have to be exhaustive. In practice, many users already treat warnings
+as errors, and switches on `bool` are rare and unidiomatic. This breaking change
+would only apply to code that has opted into the language version where this
+ships.
+
+### Warnings
+
+We don't want to mandate warnings in the language specification, but the user
+experience of a language feature is holistic and we try to design it thinking
+about the entire tooling experience. With that in mind, implementations are
+encouraged to report a static warning when:
+
```

Comment:
Perhaps introduce the notion of a 'pattern which is guaranteed to match', then mention that `_` and `T _` are always guaranteed to match, and so is a variable pattern or an identifier pattern whose type is a supertype of the static type of the matched value. With those atomic cases in place, the bullet list below can be considered as the "induction step", and they'd just talk about subpatterns as being guaranteed to match.

Reply:
I don't want to get too far into the business of specifying all of the details of exactly which patterns are or are not useless. :)